promql/functions.go

```diff
@@ -184,6 +184,137 @@ func funcIrate(ev *evaluator, args Expressions) model.Value {
 	return resultVector
 }
 
+// Calculate the smoothed value at index i in raw data d.
```

Comment:
This comment is too low-level to understand if you don't already know what the function does